src/terminal.ts

```diff
@@ -19,7 +20,17 @@ class Terminal {
     // Set holding all terminals available at any moment.
     this.terminalsSet = new Set()
 
+    // set start command asyncronously
+    // TODO does any part of the code rely on this? If so we should await the promise before there
+    setShellStartCommand().catch(() => {
+      atom.config.set("terminal.shell", getFallbackShell())
+    })
+
     this.disposables.add(
+      // an observre that checks if command has been edited manually, if so it will turn autoShell off
+      atom.config.observe("terminal.shell", () => {
```

Comment:
observe is called on activate all the time so terminal.autoShell will always be "false".

You should revert this as it has many bugs

Reply:
you should use onDidChange instead of observe

Reply:
Fixed in https://github.com/atom-ide-community/terminal/pull/27/commits/a3b67085261f747c42a591c6d153560636ea9f95